sycl/tools/sycl-prof/collector.cpp

```diff
@@ -105,27 +104,38 @@ XPTI_CALLBACK_API void xptiTraceInit(unsigned int /*major_version*/,
   }
 }
 
-XPTI_CALLBACK_API void xptiTraceFinish(const char *) { GWriter->finalize(); }
+XPTI_CALLBACK_API void xptiTraceFinish(const char *) {
+  if (GWriter != nullptr) {
+    delete GWriter;
+    GWriter = nullptr;
+  }
+}
 
 XPTI_CALLBACK_API void apiBeginEndCallback(uint16_t TraceType,
                                            xpti::trace_event_data_t *,
-                                           xpti::trace_event_data_t *,
-                                           uint64_t /*Instance*/,
+                                           xpti::trace_event_data_t *Event,
+                                           uint64_t Instance,
                                            const void *UserData) {
+  if (!GWriter)
```

Comment:
Same concern here about thread safety. Not sure if these apis (xptiTraceFinish and apiBeginEndCallback) can be called from simultaneous threads, but since this check is needed then I supposed that it is possible.